### ISSUE TITLE
Publish to AppSource does not include library app

### ISSUE THREAD

**Issue by dsaveyn** [microsoft/AL-Go#1093]
As part of the solution of #1009 the **deliverToAppSource** object was introduced in the settings. One of the properties is **includeDependencies** where you can specify the library apps.

We have configured our repo and project settings as explained [here](https://github.com/microsoft/AL-Go/blob/main/Scenarios/PublishToAppSource.md#configuration). But when publishing to appsource we receive the following error:

![image](https://github.com/microsoft/AL-Go/assets/92365904/026c9491-d6d8-445c-bc27-be37c1f53193)

Our repo looks as following:

![image](https://github.com/microsoft/AL-Go/assets/92365904/96eab779-d2db-421b-9040-129f1fa434f6)

- Core is the library app that should be included
- Manufacturing Calculation is the main app

The **AL-Go-Settings.json on repo level** looks as following:

```
{
  "type": "AppSource App",
  "templateUrl": "https://github.com/microsoft/AL-Go-AppSource@main",
  "enableCodeCop": true,
  "enableUICop": true,
  "rulesetFile": "",
  "enableExternalRulesets": true,
  "templateSha": "bf05ca89f5e08889a1d14f9a150356f141b5a632",
  "doNotRunTests": true,
  "doNotRunBcptTests": true,
  "useCompilerFolder": true,
  "doNotPublishApps": true,
  "githubrunner": "self-hosted, Windows",
  "githubRunnerShell": "powershell",
  "runs-on": "ubuntu-latest",
  "useProjectDependencies": true,
  "CICDPushBranches": [
    "main",
    "development",
    "release/*",
    "hotfix/*"
  ],
  "CICDPullRequestBranches": [
    "main",
    "development"
  ],
  "alwaysBuildAllProjects": false,
  "customCodeCops": [
    "https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.current.dll"
  ],
  "vsixFile": "latest",
  "features": [
    "TranslationFile"
  ],
  "repoVersion": "23.0",
  "versionStrategy": 23,
  "keyVaultName": "KV-AL-GO",
  "keyVaultcodesignCertificateName": "CodeSigningAlGo",
  "failOn": "warning",
  "artifact": "//23.1//latest"
}
````

The **AL-Go-Settings.json of the library app (Core)** looks as following:

```
{
  "country": "us",
  "appSourceCopMandatoryAffixes": [
    "ZESC"
  ],
  "appFolders": [],
  "testFolders": [],
  "bcptTestFolders": [],
  "installApps": ["App/.appSourceCopPackages/*.app"],
  "projectName": "Core",
  "rulesetFile": "App/.vscode/app.ruleset.json",
  "generateDependencyArtifact": true
}
```
The **AL-Go-Settings.json of the main app (Manufacturing_Calculation)** looks as following:

```
{
  "country": "us",
  "appSourceCopMandatoryAffixes": [
    "AESC"
  ],
  "appFolders": [],
  "testFolders": [],
  "bcptTestFolders": [],
  "installApps": [
    "App/.appSourceCopPackages/*.app"
  ],
  "projectName": "Manufacturing_Calculation",
  "rulesetFile": "App/.vscode/app.ruleset.json",
  "deliverToAppSource": {
    "productId": "e5207d35-2b3e-4956-9e33-7d410e1895fe",
    "mainAppFolder": "App",
    "continuousDelivery": false,
    "includeDependencies": [
      "*"
    ]
  },
  "generateDependencyArtifact": true
}
```
I also investigate the example on [https://github.com/microsoft/bcsamples-bingmaps.appsource](https://github.com/microsoft/bcsamples-bingmaps.appsource) and noticed that **every CI/CD build creates a dependencies artifact, something that's not the case in our builds**.

**Example Repo Artifacts:**
![image](https://github.com/microsoft/AL-Go/assets/92365904/5bbf58fc-e1b0-423f-ab93-bd6f15989c99)

**Our Artifacts:**
![image](https://github.com/microsoft/AL-Go/assets/92365904/9f695fa2-742a-4699-8184-21fa21147680)

Any idea why there are no dependencies artifacts are generated? We've updated the AL-Go System Files multiple times but it makes no difference.

**Comment by freddydk** [comment]
the GenerateDependencyArtifact setting is a repository setting - and needs to be in .github/AL-Go-Settings.json instead of all the .AL-Go/settings files. I think this is what is wrong here.

**Comment by freddydk** [comment]
I agree that this is in no way clear in the docs - on the contrary.
Will fix the docs.

**Comment by dsaveyn** [comment]
@freddydk Are you sure? The setting was defined on the repo level before and then I moved it to project settings because of the error

The message also literally states that it should be on the project level:

` You need to set generateDependencyArtifact to true in Manufacturing_Calculation/.AL-Go/settings.json in order to deliver dependencies `

**Comment by freddydk** [comment]
You might be right - sorry.
Let me investigate a bit more:-)

**Comment by freddydk** [comment]
You are right - in my appsource sample, it is in the project repo file

```
  "deliverToAppSource": {
    "productId": "5fbe0803-a545-4504-b41a-d9d158112360",
    "continuousDelivery": true,
    "mainAppFolder": "BingMaps-AppSource",
    "includeDependencies": [
      "Freddy Kristiansen_*.app"
    ]
  },
  "generateDependencyArtifact": true
```

If you can include the log of a build or send a download location to me - I can have a look what goes wrong

**Comment by freddydk** [comment]
I have a repro here - there is a bug when using "useCompilerFolder" - it doesn't create the dependency artifact.

**Comment by dsaveyn** [comment]
Okay that's good news. Any timeline on when we can expect a fix?

**Comment by freddydk** [comment]
I think this is a containerhelper problem - probably later today/tomorrow.

**Comment by jngillard** [comment]
Hi Freddy:

This is still not working for me. "generateDependencyArtifact": true does not cause an artifact to be generated for dependencies.

**Comment by freddydk** [comment]
Are you on preview - else it will work when we ship v5.2 (soon)

**Comment by jngillard** [comment]
Ah. That's most likely it. I'll check if this is resolved with v5.2.
Thanks,



________________________________
